src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs

```diff
@@ -32,16 +32,14 @@ internal static class MetricItemExtensions
         private static readonly ConcurrentBag<OtlpMetrics.ScopeMetrics> MetricListPool = new();
         private static readonly Action<RepeatedField<OtlpMetrics.Metric>, int> RepeatedFieldOfMetricSetCountAction = CreateRepeatedFieldOfMetricSetCountAction();
 
-        // [abeaulieu] Overload which takes a pre-converted OtlpMetrics object.
         internal static void AddMetrics(
             this OtlpCollector.ExportMetricsServiceRequest request,
             OtlpResource.Resource processResource,
             in OtlpMetrics.Metric otlpMetric,
-            in Metric metric) // [abeaulieu] [PERF] need the original metric to set the resources stuff. Technically we could build this once and clone the message.
+            in Metric metric)
```

Comment:
Not sure what to do here, since we use the metric object in order to get the scoreMetrics object and add it to the resourceMetrics.

Reply:
Let's revisit this after everything else is addressed

Reply:
After discussing this, I think it's more efficient to keep the ArrayPool of `Metric` instead of copying meter name and meter version strings for each metric in the batch.

The only thing I would do is rename `meterNames` which is no longer representative of the the list (see my other comment).